File: src/main/java/me/testaccount666/serversystem/commands/executables/disposal/CommandDisposal.java
package me.testaccount666.serversystem.commands.executables.disposal;

import me.testaccount666.serversystem.commands.ServerSystemCommand;
import me.testaccount666.serversystem.commands.executables.AbstractServerSystemCommand;
import me.testaccount666.serversystem.managers.PermissionManager;
import me.testaccount666.serversystem.userdata.User;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryType;

import static me.testaccount666.serversystem.utils.MessageBuilder.command;
import static me.testaccount666.serversystem.utils.MessageBuilder.general;

@ServerSystemCommand(name = "disposal")
public class CommandDisposal extends AbstractServerSystemCommand {

    @Override
    public void execute(User commandSender, Command command, String label, String... arguments) {
        if (!checkBasePermission(commandSender, "Disposal.Use")) return;
        if (handleConsoleWithNoTarget(commandSender, getSyntaxPath(command), label, arguments)) return;

        var targetUserOptional = getTargetUser(commandSender, arguments);
        if (targetUserOptional.isEmpty()) {
            general("PlayerNotFound", commandSender).target(arguments[0]).build();
            return;
        }

        var targetUser = targetUserOptional.get();
        var targetPlayer = targetUser.getPlayer();
        var isSelf = targetUser == commandSender;

        if (!isSelf && !checkOtherPermission(commandSender, "Disposal.Other", targetPlayer.getName())) return;

        targetPlayer.openInventory(Bukkit.createInventory(targetPlayer, InventoryType.CHEST.getDefaultSize() * 2, "§cTrash"));

        if (isSelf) return;
        command("Disposal.Success", targetUser).target(targetPlayer.getName()).build();
    }

    @Override
    public String getSyntaxPath(Command command) {
        return "Disposal";
    }

    @Override
    public boolean hasCommandAccess(Player player, Command command) {
        return PermissionManager.hasCommandPermission(player, "Disposal.Use", false);
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/commands/executables/disposal/CommandDisposal.kt
package me.testaccount666.serversystem.commands.executables.disposal

import me.testaccount666.paperktx.extensions.ComponentExtensions.asComponent
import me.testaccount666.serversystem.commands.ServerSystemCommand
import me.testaccount666.serversystem.commands.executables.AbstractServerSystemCommand
import me.testaccount666.serversystem.extensions.commandMsg
import me.testaccount666.serversystem.extensions.generalMsg
import me.testaccount666.serversystem.userdata.User
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.event.inventory.InventoryType

@ServerSystemCommand("disposal")
class CommandDisposal : AbstractServerSystemCommand() {
    override fun getUsagePermission(command: Command) = "Disposal.Use"
    override fun getSyntaxPath(command: Command?) = "Disposal"

    override fun execute(commandSender: User, command: Command, label: String, vararg arguments: String) {
        if (isConsoleWithNoTarget(commandSender, getSyntaxPath(command), label, arguments = arguments)) return

        val targetUser = getTargetUser(commandSender, arguments = arguments) ?: run {
            commandSender.generalMsg("PlayerNotFound") { target(arguments[0]) }
            return
        }

        val targetPlayer = targetUser.getPlayer()!!
        val isSelf = targetUser === commandSender

        if (!isSelf && !checkPermission(commandSender, "Disposal.Other", targetPlayer.name)) return

        val inventory = Bukkit.createInventory(targetPlayer, InventoryType.CHEST.defaultSize * 2, "&cTrash".asComponent())

        targetPlayer.openInventory(inventory)

        if (isSelf) return
        targetUser.commandMsg("Disposal.Success") { target(targetPlayer.name) }
    }
}
